conan/tools/files/patches.py

```diff
@@ -88,6 +88,9 @@ def apply_conandata_patches(conanfile):
     if isinstance(patches, dict):
         assert conanfile.version, "Can only be applied if conanfile.version is already defined"
         entries = patches.get(str(conanfile.version), [])
+        if entries is None:
+            conanfile.output.info("apply_conandata_patches(): No patches defined for version in conandata")
```

Comment:
```suggestion
            conanfile.output.warning(f"apply_conandata_patches(): No patches defined for version {conanfile.version} in conandata.yml")
```

Reply:
@RubenRBS I am not that sure this needs to be a warning. It is not that infrequent that there are packages that applies patches in some versions but not others, and this is a total valid use case, so this could be noisy/false-positive for those versions?

Reply:
Oh, nevermind, sorry I understood it incorrectly. A warning makes sense.